grammar.js

```diff
@@ -530,13 +557,40 @@ module.exports = grammar({
                 /\n\s*/
             ),
 
+        // Definition:
+        // ---@field [public|protected|private] field_name MY_TYPE[|other_type] [@comment]
+        //
+        // I don't think [public|protected|private] is useful for us.
+        //
+        // ---@field example table hello
+        // ---@field example (table): hello
+        emmy_field: ($) =>
+            seq(
+                /\s*---@field\s+/,
+                field("name", $.identifier),
+                field("type", list_of($.emmy_type, /\s*\|\s*/)),
+
+                // TODO: How closely should we be to emmy...
+                optional(
+                    seq(
+                        /\s*:\s*/,
+                        field("description", $.field_description)
+                    )
+                ),
+                /\n\s*/
+            ),
+
         _multiline_emmy_string: ($) =>
             prec.right(
                 PREC.PRIORITY,
                 seq(/[^\n]+/, any_amount_of(/\s*---[^\n]*/))
                 // seq(/[^\n]*/, any_amount_of(/\n\s*---[^\n]*/))
             ),
 
+        class_description: ($) => $._multiline_emmy_string,
+        field_description: ($) => $._multiline_emmy_string,
+
+        // TODO(conni2461): Pretty sure that doesn't work as expected
```

Comment:
I tried to fix this `_multiline_emmy_string` thing because its not multiline but it just doesn't want to work.
I can either make it multiline but than it will no longer pickup the next annotation

Reply:
I have had this problem as well. Maybe should write it in the scanner.cc